src/app/utils/utils.js

```diff
@@ -368,6 +370,60 @@ function getAggregatedElectronicResources(items = []) {
   return _flatten(electronicResources);
 }
 
+/**
+ * Return a list of non Aeon Link Electronic Resouces from given list
+ * @param {array} resources Resources[ ]
+ * @param {array} items Items[ ]
```

Comment:
Love to see this documentation! Isn't the type of `resources` and `items` actually the same, though? The former is the set of _electronic_ items aggregated out earlier whereas the latter is the complete set of items for the bib?

Reply:
Glad you like it. 

TLDR: They are not the same. I am adding a commit with a small addition defining a simple interface for these two parameters. 

> The former is the set of electronic items aggregated out earlier whereas the latter is the complete set of items for the bib?

The above question is correct. 

I understand why it's presumed the two objects passed into this function are the same, however they are not. The "Resources" comes from the list of "Items", but they are two different objects. Starting with [line 98](https://github.com/NYPL/discovery-front-end/blob/4d6023f2b498a736715b817dd3860c8be5fdf48a/src/app/pages/BibPage.jsx#L98) on the [BibPage](https://github.com/NYPL/discovery-front-end/blob/development/src/app/pages/BibPage.jsx) component you can see the `item` variable being instantiated; [line104](https://github.com/NYPL/discovery-front-end/blob/4d6023f2b498a736715b817dd3860c8be5fdf48a/src/app/pages/BibPage.jsx#L104) shows the `aggregatedElectronicResources` variable being instantiated.

The `resources` are extracted from the `item` and the two objects do not == || ===

If you're perhaps suggesting this function to accept only the `item` as a parameter and to then extract the `aggregatedElectronicResources` from it, I'd argue against that. 

1. The logic to extract the resources has already been done. 
2. It will add unnecessary additional logical complexity to this function. 
3. It changes the purpose/definition of this function.

Hope this answers your question and clears things up.

Reply:
I see. My question was actually about the _type_ of `resources` and `items`. I understand they're different in terms of `===` but I thought they were both arrays of the same type. I see now that `aggregatedElectronicResources` is actually an array of items that have been [mapped via a getElectronicResources method, which only includes a specific item property](https://github.com/NYPL/discovery-front-end/blob/4d6023f2b498a736715b817dd3860c8be5fdf48a/src/app/utils/item.js#L163), so the _type_ of objects in the `resources` array is different from the _type_ of objects in the `items` array. The added comments help.